src/datasets/builder.py

```diff
@@ -764,6 +784,19 @@ def as_dataset(
 
         Returns:
             datasets.Dataset
+
+        Example:
+
+        ```py
+        >>> from datasets import load_dataset_builder
+        >>> builder = load_dataset_builder('rotten_tomatoes')
+        >>> ds = builder.as_dataset(split='train')
```

Comment:
we usually download and prepare the dataset before opening it :p
```suggestion
        >>> builder = load_dataset_builder('rotten_tomatoes')
        >>> ds = builder.download_and_prepare()
        >>> ds = builder.as_dataset(split='train')
```